node/subsystem-util/src/database.rs

```diff
@@ -179,58 +175,47 @@ pub mod paritydb_impl {
 			map_err(self.db.get(col as u8, key))
 		}
 
-		fn get_by_prefix(&self, col: u32, prefix: &[u8]) -> Option<Box<[u8]>> {
-			self.iter_with_prefix(col, prefix).next().map(|(_, v)| v)
+		fn get_by_prefix(&self, col: u32, prefix: &[u8]) -> Result<Option<DBValue>> {
+			self.iter_with_prefix(col, prefix)
+				.next()
+				.transpose()
+				.map(|mb| mb.map(|(_, v)| v))
 		}
 
-		fn iter<'a>(&'a self, col: u32) -> Box<dyn Iterator<Item = (Box<[u8]>, Box<[u8]>)> + 'a> {
-			let mut iter = handle_err(self.db.iter(col as u8));
+		fn iter<'a>(&'a self, col: u32) -> Box<dyn Iterator<Item = Result<DBKeyValue>> + 'a> {
+			let mut iter = match self.db.iter(col as u8) {
+				Ok(iter) => iter,
+				Err(e) => return Box::new(std::iter::once(map_err(Err(e)))),
+			};
 			Box::new(std::iter::from_fn(move || {
```

Comment:
```suggestion
			Box::new(iter.map(|r| map_err(r.map(|(k, v)| (k.into(), v))))
```

That should do the same and is easier to read?

Reply:
This doesn't work as parity-db impl doesn't implement iterator, but only [`next`[(https://github.com/paritytech/parity-db/blob/43b451edfdb85e463ab181dd0951872c29b863ac/src/btree/iter.rs#L304) method.
@cheme correct me if I'm wrong

Reply:
Ahh okay...